addons/website_slides/models/slide_slide.py

```diff
@@ -433,6 +535,20 @@ def write(self, values):
             self.date_published = datetime.datetime.now()
             self._post_publication()
 
+        # avoid fetching external metadata when installing the module (i.e. for demo data)
+        # we also support a context key if you don't want to fetch the metadata when modifying a slide
+        if any(values.get(url_param) for url_param in ['url', 'video_url', 'document_google_url', 'image_google_url']) \
+           and not self.env.context.get('install_mode', False) \
+           and self.env.context.get('website_slides_fetch_metadata', True):
+            slide_metadata = self._fetch_external_metadata()
+            if slide_metadata:
+                # only update keys that are not set in the incoming values and for which we don't have a value yet
+                self.update({
+                    key: value
+                    for key, value in slide_metadata.items()
+                    if key not in values.keys() and not any(slide[key] for slide in self)
```

Comment:
Same here: we could call _fetch_external_metadata only if some fields are missing. Purpose would be to speedup if possible when we already have some information.